### PR TITLE
telemetry: add UI with default opt-out

### PR DESCRIPTION
**Pull request by ericpromislow**
The actual telemetry should be done in a separate PR, as I don't think we can follow `dashboard`'s lead here, as our execution model is different.

What to look for right now:

There's a new checkbox on the System Prefs page, along with a popup icon to the right of the checkbox label.
And yeah, I'm not stoked that this checkbox is on the System Prefs page. Should we create a separate `Settings` page?

Signed-off-by: Eric Promislow <epromislow@suse.com>

### CODE REVIEW

File: src/components/SystemPreferences.vue [hunk 1]
```diff
@@ -39,6 +40,11 @@ export default {
       type:    Number,
       default: 0,
     },
+    // misc
+    telemetry: {
+      type:    Boolean,
+      default: false,
```
Comment (mook-as) [inline src/components/SystemPreferences.vue:46]: #26 wants opt-out, not opt-in. (Personally I'd much prefer opt-in, of course, but that's a PM battle.)

File: src/components/SystemPreferences.vue [hunk 2]
```diff
@@ -179,6 +188,26 @@ export default {
...
+    <div class="checkbox">
+      <Checkbox
+        :value="telemetry"
+        label="Allow collection of anonymous statistics to help us improve Rancher Desktop"
+        @input="toggleTelemetry"
+      />
+      <v-popover placement="right">
+        <i class="icon icon-info" />
+        <span slot="popover">
+          Rancher Labs would like to collect a bit of anonymized information<br />
```
Comment (mook-as) [inline src/components/SystemPreferences.vue:202]: There doesn't seem to be any reason to have explicit `<br>`s here; HTML can do the layout just fine.

Are we still referring to ourselves as Rancher Labs here, post-acquisition?

File: src/components/SystemPreferences.vue [hunk 3]
```diff
@@ -179,6 +188,26 @@ export default {
...
+          about the configuration of your installation to help make Rancher Desktop better.<br /><br />
```
Comment (mook-as) [inline src/components/SystemPreferences.vue:203]: `<br><br>` should just be done with proper `<p>`s instead; it's actually semantic too (they're actually paragraphs).

File: src/components/SystemPreferences.vue [hunk 4]
```diff
@@ -179,6 +188,26 @@ export default {
...
+          Your data will not be shared with anyone else, and no information about<br />
+          what specific resources or endpoints you are deploying is included.<br />
+          Once enabled you can view exactly what data will be sent at <code>/v1-telemetry</code>.<br /><br />
+          <a href="https://rancher.com/docs/rancher/v2.x/en/faq/telemetry/" target="_blank">More Info</a>
```
Comment (mook-as) [inline src/components/SystemPreferences.vue:207]: This needs to open in the user's [default browser](https://www.electronjs.org/docs/api/shell#shellopenexternalurl-options), _not_ in Electron.  We may need to do that in the main process on [an event](https://www.electronjs.org/docs/api/web-contents#event-will-navigate) for all `http` links.

File: src/components/form/Checkbox.vue [hunk 5]
```diff
@@ -1,31 +1,33 @@
 <script>
-import { _EDIT } from '@/config/query-params';
```
Comment (mook-as) [inline src/components/form/Checkbox.vue:2]: Please put this in a separate commit (I assume this is just straight copied from dashboard).

File: src/config/settings.ts [hunk 6]
```diff
@@ -23,6 +24,7 @@ const defaultSettings = {
     numberCPUs:  2,
   },
   portForwarding: { includeKubernetesServices: false },
+  telemetry:      false,
```
Comment (mook-as) [inline src/config/settings.ts:27]: See above re: defaults.

**Review by mook-as — CHANGES_REQUESTED** [review]
The tooltip looks really weird here; is it something local? Note also the scrollbars popping up (which only showed up once I tried to invoke the tooltip).  I think it's because `v-tooltip` also creates a `<div class="wrapper">`?

image

**Review by mook-as — DISMISSED** [review]
Screenshot of what it looks like (on my screen):
image

I find that this, as the _only_ thing in the welcome screen, to be a bit intimidating.  The previous implementation with the tooltip (once we fix the CSS issues) would probably be better… but then I'm not a UI person, so don't trust what I say :)

I'm okay with the code changes (though I'd like to run a cleanup PR on some of the adjacent stuff after), but I'd like @mattfarina to approve the UI before we merge.

**Comment by mattfarina** [comment]
@ericpromislow can you fix the merge conflict?

**Comment by mattfarina** [comment]
@mook-as I have a now old PR to rewrite the welcome screen. I'll be refreshing tomorrow.

File: src/components/form/Checkbox.vue (hunk 5 above)
Comment (ericpromislow) [reply src/components/form/Checkbox.vue]: done in commit 79fe043

File: src/components/SystemPreferences.vue (hunk 1 above)
Comment (ericpromislow) [reply src/components/SystemPreferences.vue]: that tooltip is obsolete

File: src/components/SystemPreferences.vue (hunk 4 above)
Comment (ericpromislow) [reply src/components/SystemPreferences.vue]: This code was dropped.

File: src/components/SystemPreferences.vue (hunk 3 above)
Comment (ericpromislow) [reply src/components/SystemPreferences.vue]: done -- I had copied this from dashboard, but, yeah, prefer the better markup

File: src/components/TelemetryOptIn.vue [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        Send anonymized usage info, error reports, etc. to help improve Rancher Desktop. Your data will not be shared with anyone else, and no information about what specific resources or endpoints you are deploying is included.
+      </p>
+    </div>
+  </div>
+</template>
+
+<style scoped>
+p.fineprint {
+  font-size: small;
+  margin-right: 50%;
```
Comment (mook-as) [inline src/components/TelemetryOptIn.vue:39]: That's… odd, but okay.

**Review by mook-as — APPROVED** [review]
image

**Comment by mattfarina** [comment]
Merging as I will be re-doing this pane shortly and tweaking the layout anyway.
